src/datasources/auth-api/utils/__tests__/to-signable-siwe-message.spec.ts

```diff
@@ -0,0 +1,328 @@
+      const message = siweMessageBuilder()
+        .with('scheme', undefined)
+        .with('statement', undefined)
+        .with('expirationTime', faker.date.recent().toISOString())
+        .with('notBefore', undefined)
+        .with('requestId', undefined)
+        .with('resources', undefined)
+        .build();
+
+      const result = toSignableSiweMessage(message);
+
+      expect(result)
+        .toBe(`${message.domain} wants you to sign in with your Ethereum account:
+${message.address}
+URI: ${message.uri}
+Version: ${message.version}
+Chain ID: ${message.chainId}
+Nonce: ${message.nonce}
+Issued At: ${message.issuedAt}
+Expiration Time: ${message.expirationTime}`);
+    });
+
+    it('should not add an empty expirationTime', () => {
+      const message = siweMessageBuilder()
+        .with('scheme', undefined)
+        .with('statement', undefined)
+        .with('expirationTime', '')
+        .with('notBefore', undefined)
+        .with('requestId', undefined)
+        .with('resources', undefined)
+        .build();
+
+      const result = toSignableSiweMessage(message);
+
+      expect(result)
+        .toBe(`${message.domain} wants you to sign in with your Ethereum account:
+${message.address}
+URI: ${message.uri}
+Version: ${message.version}
+Chain ID: ${message.chainId}
+Nonce: ${message.nonce}
+Issued At: ${message.issuedAt}`);
+    });
+  });
+
+  describe('notBefore', () => {
+    it('should add the notBefore time if present', () => {
+      const message = siweMessageBuilder()
+        .with('scheme', undefined)
+        .with('statement', undefined)
+        .with('expirationTime', undefined)
+        .with('notBefore', faker.date.recent().toISOString())
+        .with('requestId', undefined)
+        .with('resources', undefined)
+        .build();
+
+      const result = toSignableSiweMessage(message);
+
+      expect(result)
+        .toBe(`${message.domain} wants you to sign in with your Ethereum account:
+${message.address}
+URI: ${message.uri}
+Version: ${message.version}
+Chain ID: ${message.chainId}
+Nonce: ${message.nonce}
+Issued At: ${message.issuedAt}
+Not Before: ${message.notBefore}`);
+    });
+
+    it('should not add an empty notBefore', () => {
+      const message = siweMessageBuilder()
+        .with('scheme', undefined)
+        .with('statement', undefined)
+        .with('expirationTime', undefined)
+        .with('notBefore', '')
+        .with('requestId', undefined)
+        .with('resources', undefined)
+        .build();
+
+      const result = toSignableSiweMessage(message);
+
+      expect(result)
+        .toBe(`${message.domain} wants you to sign in with your Ethereum account:
+${message.address}
+URI: ${message.uri}
+Version: ${message.version}
+Chain ID: ${message.chainId}
+Nonce: ${message.nonce}
+Issued At: ${message.issuedAt}`);
+    });
+  });
+
+  describe('requestId', () => {
+    it('should add the requestId if present', () => {
```

Comment:
Nitpick: I see the value on having tests verifying the absence of these optional fields, but the case where they are present should be convered by the first one (`should return a signable message with all fields`). Maybe we could get rid of some of them, wdyt? No strong opinion though, I left this totally up to you, if you see value on having them I'm totally OK with it.

Reply:
I'd be in favour of keeping them. I wanted to remain as explicit as possible because it adheres to the [message field standards](https://eips.ethereum.org/EIPS/eip-4361#message-fields) (we should ideally be using the [official package](https://www.npmjs.com/package/siwe) - reasons for not are explained in the description).